### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by NateMoeller**

### CODE REVIEW

File: api/src/main/java/com/nicknathanjustin/streamercontracts/donations/DonationsApiController.java [hunk 1]
```diff
@@ -53,6 +59,16 @@ public ResponseEntity createDonation(@RequestBody @NonNull final CreateDonationR
 
         final ContractModel contract = contractService.createContract(proposer, streamer, game, createDonationRequest.getBounty());
         donationService.createDonation(contract, proposer, createDonationRequest.getAmount(), contract.getProposedAt(), paypalPaymentId, authorizationId);
+
+        // send notification to FE
```
Comment (nckackerman) [inline api/src/main/java/com/nicknathanjustin/streamercontracts/donations/DonationsApiController.java]: optional: remove comment and wrap logic in a self documenting function

```
private final alertFrontEnd() {
//logic
}
```

File: web/src/app/notification/Notification.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      return this.getItem(notification.title, notification.description, index);
+    });
+    const content = items.length > 0 ? items : <div className={styles.noNotifications}>No notifications</div>;
+
+    const popover = (
+      <Popover id="popover-positioned-bottom">
+        {content}
+      </Popover>
+    );
+    const number = this.props.notification.notifications.length > 99 ? '99+' : this.props.notification.notifications.length;
```
Comment (nckackerman) [inline web/src/app/notification/Notification.js]: super nit: pulling 99 into a const variable is slightly cleaner

File: api/src/main/java/com/nicknathanjustin/streamercontracts/alerts/AlertServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.messaging.simp.SimpMessageSendingOperations;
+
+@RequiredArgsConstructor
+public class AlertServiceImpl implements AlertService {
+
+    @NonNull final private SimpMessageSendingOperations messagingTemplate;
+
+    @Override
+    public void sendAlert(@NonNull final String alertChannelId, @NonNull final String title, @NonNull final String description) {
+        final AlertMessage testMessage = new AlertMessage(title, description);
```
[the remark this answers is not in the document]
Comment (m1sterc001guy) [reply api/src/main/java/com/nicknathanjustin/streamercontracts/alerts/AlertServiceImpl.java]: My guess is that it was copied from above?

File: api/src/main/java/com/nicknathanjustin/streamercontracts/donations/DonationsApiController.java [hunk 4]
```diff
@@ -53,6 +59,16 @@ public ResponseEntity createDonation(@RequestBody @NonNull final CreateDonationR
...
+        final String title = "New bounty from " + contract.getProposer().getTwitchUsername();
```
Comment (m1sterc001guy) [inline api/src/main/java/com/nicknathanjustin/streamercontracts/donations/DonationsApiController.java]: Nit: Make "New bounty from %s" a constant static field and reference in both controllers

**Pull request merged**